libs/core/hardware/include/hpx/hardware/timestamp/linux_generic.hpp

```diff
@@ -12,13 +12,23 @@
 
 #include <time.h>
 
+#include <hpx/config.hpp>
+
+#if define(HPX_HAVE_CUDA)
+#include <hpx/hardware/timestamp/cuda.hpp>
+#endif
+
 namespace hpx { namespace util { namespace hardware {
 
-    inline std::uint64_t timestamp()
+    HPX_HOST_DEVICE std::uint64_t timestamp()
     {
+#if defined(HPX_HAVE_CUDA) && defined(__CUDA_ARCH__)
+        return timestamp_cuda();
```

Comment:
Is there any reason not to add this to the bgq and msvc files? We don't test them with CUDA, but I don't see any reason why it shouldn't work there (modulo compiler bugs).

Reply:
I haven't tested them with bgq or msvc. So, I decided to play safe and not add it there.